.github/workflows/pull_requests.yml

```
@@ -0,0 +1,31 @@
+name: Pull Request
+
+on:
+  pull_request:
+    branches: [ main ]
```

Might be a good idea to keep running tests on the main branch as well, that'll also enable the repo to have those "CI badge"s.
In that case you'll need to rename this file and change the name of the CI.

I'm also not sure if this'll run in any PRs since no PR is based on the "main" branch.
As you can see it's not running in this PR either, while it should.

[In Vapor](https://github.com/vapor/vapor/blob/ebbe71c89aa1b76a0920277760b12be7a2ec7c70/.github/workflows/test.yml#L6), it looks like this:

```suggestion
  pull_request: { types: [opened, reopened, synchronize, ready_for_review] }
  push: { branches: [ main ] }
```
The `pull_request` part will avoid running tests in e.g. draft pull requests.
Not sure if that's desirable, but Vapor has been using this.

---

I intentionally only focused on pull request based triggers for now. This is the first of many PRs to setup or GH action but I did not want to add all at once. Thanks for the trigger suggestion. Let's try it out!